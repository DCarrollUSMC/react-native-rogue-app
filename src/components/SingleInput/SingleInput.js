/*
 * FILENAME:        components/SingleInput/SingleInput.js
 *
 * DESCRIPTION:     Individual user input field component with optional image
 *
 * Author: SixPack Tech Team
 * Copyright (c) 2019 - Six-Pack Labs
 */

import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Dimensions from 'Dimensions';
import {StyleSheet, View, TextInput, Text, Image} from 'react-native';
import { defined } from '../../app/helpers';

export default class SingleInput extends Component {

    render() {
        const iconStyle = {
            width: this.props.iconWidth,
            height: this.props.iconHeight,
        }
        const txtBoxStyle = {
            color: this.props.txtColor,
            backgroundColor: this.props.bgColor,
            borderBottomColor: this.props.borderBottomColor,
            borderBottomWidth: 1
        }

        return (
            <View style={styles.inputOuterWrapper}>
                <View style={styles.inputWrapper}>
                    {this.props.source &&
                        <Image source={this.props.source} style={[styles.inlineImg, iconStyle]} />
                    }
                    <TextInput
                    value={this.props.value}
                    style={[styles.input, txtBoxStyle]}
                    placeholder={this.props.placeholder}
                    secureTextEntry={this.props.secureTextEntry}
                    autoCorrect={this.props.autoCorrect}
                    autoCapitalize={this.props.autoCapitalize}
                    returnKeyType={this.props.returnKeyType}
                    placeholderTextColor={this.props.placeholderTextColor}
                    underlineColorAndroid='transparent'
                    onChangeText={this.props.onChangeText}
                    />
                </View>
                {defined(this.props.error, false, true) &&
                    <Text style={{color: 'red'}}>{this.props.error}</Text>
                }
            </View>
        );
    }
}

SingleInput.propTypes = {
    source: PropTypes.number,
    placeholder: PropTypes.string.isRequired,
    secureTextEntry: PropTypes.bool,
    autoCorrect: PropTypes.bool,
    autoCapitalize: PropTypes.string,
    returnKeyType: PropTypes.string,
    value: PropTypes.string,
    iconWidth: PropTypes.number,
    iconHeight: PropTypes.number,
    bgColor: PropTypes.string,
    txtColor: PropTypes.string,
    placeholderTextColor: PropTypes.string,
    borderBottomColor: PropTypes.string,
    error: PropTypes.string
};

const DEVICE_WIDTH = Dimensions.get('window').width;

const styles = StyleSheet.create({
    input: {
        width: DEVICE_WIDTH - 40,
        height: 50,
        marginHorizontal: 10,
        paddingLeft: 65,
        borderRadius: 5,
    },
    inputOuterWrapper: {
        justifyContent: 'center',
        alignItems: 'center',
        marginTop:5
    },
    inputWrapper: {
        justifyContent: 'center',
        alignItems: 'center',
    },
    inlineImg: {
        position: 'absolute',
        zIndex: 99,
        left: 35
    },
});