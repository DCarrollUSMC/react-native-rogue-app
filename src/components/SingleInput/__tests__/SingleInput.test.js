/*
 * FILENAME:        components/SingleInput/__tests__/SingleInput.test.js
 *
 * DESCRIPTION:     Test file for SingleInput Component
 * 
 * Author: SixPack Tech Team
 * Copyright (c) 2019 - Six-Pack Labs
 */

import React from 'react';
import renderer from 'react-test-renderer';

import SingleInput from '../SingleInput';

describe('SingleInput', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <SingleInput 
                placeholder='Test Input'
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});