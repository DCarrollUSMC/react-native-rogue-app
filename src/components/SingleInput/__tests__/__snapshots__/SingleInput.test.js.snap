// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleInput renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="Test Input"
      style={
        Array [
          Object {
            "borderRadius": 5,
            "height": 50,
            "marginHorizontal": 10,
            "paddingLeft": 65,
            "width": 710,
          },
          Object {
            "backgroundColor": undefined,
            "borderBottomColor": undefined,
            "borderBottomWidth": 1,
            "color": undefined,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
